src/openrct2/ride/Vehicle.cpp

```diff
@@ -6600,26 +6600,34 @@ static void vehicle_update_track_motion_up_stop_check(rct_vehicle * vehicle)
 }
 
 /**
- * Modifies the train's velocity to match the block-brake fixed velocity.
+ * Modifies the train's velocity to match the block-brakes specified velocity.
  * This function must be called when the car is running through a non-stopping
  * state block-brake (precondition), which means that the block brake is acting
  * merely as a velocity regulator, in a closed state. When the brake is open, it
- * boosts the train to the speed limit
+ * boosts the train to the fixed block release speed.
  */
-static void apply_non_stop_block_brake(rct_vehicle * vehicle, bool block_brake_closed)
+static void apply_non_stop_block_brake(rct_vehicle * vehicle, bool block_brake_closed, int32_t speed)
 {
+    if (speed < BLOCK_BRAKE_BASE_SPEED)
+    {
+        //Fix blocks that were saved before block section speed settings were implemented.
+        speed = BLOCK_BRAKE_BASE_SPEED;
```

Comment:
When the block is closed it behaves exactly the same as stock. There is no situation where the train could pass now that it couldn't already do it stock rct2.

Reply:
Ratings are calculated based on the first test run. Subsequent runs do not affect the rating. This is the case currently as well.

Reply:
Added a fix for the 0 speed issue. it will now display 4mph or 6km/h like it should.

Reply:
The reason it doesn't speed it up is because that is not how block brakes work in real life. Rides must be designed that they can make it through the course even after an emergency stop on the block. Also you should design your ride so that it shouldn't need to stop on any of the blocks in main part of your course under normal circumstances anyway.